rasterio/plot.py

```diff
@@ -105,31 +105,33 @@ def show(source, with_bounds=True, contour=False, contour_label_kws=None,
                 # will have the colors in the order expected by
                 # matplotlib (RGB)
                 arr = source.read(rgb_indexes, masked=True)
-                arr = reshape_as_image(arr)
 
             except KeyError:
-                arr = source.read(1, masked=True)
+                arr = source.read((3,2,1), masked=True)
```

Comment:
why do you assume that bands are store as BGR ?

Reply:
This is from our experience in remote sensing application, that is, if the remote sensing image contains multiple bands, the bands are usually stored by ascending order of the wavelength of bands. The BGR wavelengths are: ～460nm, ~530nm, ~580nm, therefore for a multibands image we set the default index (3,2,1) for the color composition, thus mostly the color composition is the true color composition rather than by setting the default index of (1,2,3). 
moreover, if the image bands are not stored as BGR, our revision also at least show the multibands image as a color composite image,  while the previous script only show the multibands image as a grey image.

Reply:
we add a new feature for image histogram stretching,  this feature is used by an optional parameter for users, which will make the image visualization better. i frequently use it for image visualization in my experiment.

Reply:
> This is from our experience in remote sensing application, that is, if the remote sensing image contains multiple bands, the bands are usually stored by ascending order of the wavelength of bands. The BGR wavelengths are: ～460nm, ~530nm, ~580nm, therefore for a multibands image we set the default index (3,2,1) for the color composition, thus mostly the color composition is the true color composition rather than by setting the default index of (1,2,3).

This is a pretty heavy assumption... which after 15years working with Remote Sensing data, I would not make myself ;-) 

>  if the image bands are not stored as BGR, our revision also at least show the multibands image as a color composite image, while the previous script only show the multibands image as a grey image.

Which is why I think adding an `indexes=` options might be a better solution

Reply:
that is also ok, we add a new 'indexes=' options, and the user can specify the bands for color composite when visualize the multibands image. Hope you can check my code again, thanks.